hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -251,9 +251,15 @@ public void renameTable(TableIdentifier from, TableIdentifier originalTo) {
     } catch (NoSuchObjectException e) {
       throw new NoSuchTableException("Table does not exist: %s", from);
 
-    } catch (AlreadyExistsException e) {
-      throw new org.apache.iceberg.exceptions.AlreadyExistsException(
-          "Table already exists: %s", to);
+    } catch (RuntimeException e) {
```

Comment:
You should catch here `MetaException` instead of `RuntimeException`  from the altertable and throw `AlreadyExistsException` ?

Reply:
or if it is not coming as `MetaException`, if it is coming as `InvalidOperationException` cause. We should unwrap this at alterTable and throw only the cause.

Reply:
Unwrapping only `MetaException` looks odd.

Reply:
I am not sure why we should explicitly catch MetaException here. Since it's already being handled with 
```
catch (TException e) {
      throw new RuntimeException("Failed to rename " + from + " to " + to, e);

    }
```

`InvalidOperationException` will always be wrapped with `RuntimeException` in case of alterTable. Either we should unwrap it `MetastoreUtil` or at every caller of it.

Reply:
Originally we were unwrapping to `MetaException` but here were still catching `RuntimeException`. Hence I gave that comment. 

Now we are unwrapping to `TException` and catching the `TException`. So, now LGTM.